libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/marker/LongPressMapComponent.kt

```diff
@@ -25,7 +28,15 @@ internal class LongPressMapComponent(
     }
 
     private val longClickListener = OnMapLongClickListener { point ->
-        context.dispatch(DestinationAction.SetDestination(Destination(point)))
+        context.viewModel.destinationViewModel.invoke(
+            DestinationAction.SetDestination(Destination(point))
+        )
+        context.viewModel.routesViewModel.invoke(
+            RoutesAction.SetRoutes(emptyList())
+        )
+        context.viewModel.navigationStateViewModel.invoke(
+            NavigationStateAction.Update(NavigationState.DestinationPreview)
+        )
```

Comment:
With the changes above, it alters the behavior from what we had in the main branch

| Before(main branch) | After(this branch) |
| -- | -- |
| https://user-images.githubusercontent.com/9770186/159797899-fa3210c6-30d3-407d-a7f8-45dc01677435.mp4 | https://user-images.githubusercontent.com/9770186/159797937-dd0b1c32-806c-4496-ac1e-d384c296ac76.mp4 |

Reply:
Maybe I can fix this in the CameraComponent now

Reply:
Ah after talking i understand the issue. 

When in RoutePreview, the long press should be requesting new routes. The issue was, the long press was clearing the route and only updating the marker.

To handle these two experiences, we split the LongPressMapComponent into 2 components
``` kotlin
FreeDriveLongPressMapComponent
RoutePreviewLongPressMapComponent
```